src/stactools/sentinel2/stac.py

```diff
@@ -364,8 +373,17 @@ def set_asset_properties(_asset: pystac.Asset, _band_gsd: Optional[int] = None):
         viewing_angle = viewing_angles[band_id]
         # We can't use the ViewExtension here until
         # https://github.com/stac-utils/pystac/issues/793 is fixed
-        asset.extra_fields["view:azimuth"] = viewing_angle.azimuth
-        asset.extra_fields["view:incidence_angle"] = viewing_angle.zenith
+        if not math.isnan(viewing_angle.azimuth):
```

Comment:
I filed https://github.com/stac-extensions/view/issues/4 to discuss whether these fields should be allowed by the spec on Assets